src/RVRP_data.jl

```diff
@@ -1,118 +1,81 @@
-    time_windows::Vector{TimeWindow} # optional
-    duration::Float64 # optional
-    req_id::String
-end
-
-struct Operation <: AbstractNode
-    start_location::Location
-    end_location::Location
-    required_capacity::Float64
-    delta_capacity::Float64
-    time_windows::Vector{TimeWindow} # optional
-    duration::Float64 # optional      
-    req_id::String
-end
-
-abstract type AbstractRequest end
-
-struct PickupRequest <: AbstractRequest
-    id::String
-    node::Pickup
-end    
-
-struct DeliveryRequest <: AbstractRequest
-    id::String
-    node::Delivery
-end    
-
-struct OperationRequest <: AbstractRequest
-    id::String
-    node::Operation
-end    
-
-struct Shipment <: AbstractRequest
-    id::String
-    pickup::Pickup
-    delivery::Delivery
-end
-
-abstract type AbstractVehicle end
-
-struct Vehicle <: AbstractVehicle
-    id::String
-    depot::Depot # optional
-    v_type::VehicleType
-    time_schedule::TimeWindow
-    return_to_depot::Bool
-    infinite_copies::Bool
-    initial_load::Float64
-    picked_shipments::Vector{Shipment}
-end
-
-struct RvrpProblem
-    problem_id::String
-    problem_type::ProblemType
-    vehicles::Vector{Vehicle}
-    vehicle_types::Vector{VehicleType}
-    distance_matrix::Array{Float64,2}
-    travel_times_matrix::Array{Float64,2}
-    # Requests
-    pickups::Vector{PickupRequest}
-    deliveries::Vector{DeliveryRequest}
-    operations::Vector{OperationRequest}
-    shipments::Vector{Shipment}
-    picked_shipments::Vector{Shipment}
-end 
+abstract type AbstractNode end
+
+struct ProblemType
+    fleet_size::String # INFINITE or FINITE
+    fleet_composition::String # HOMOGENEOUS or HETEROGENEOUS
+end
+
+struct Coord
+    x::Float64
+    y::Float64
+end
+
+struct Location # define: index + distance matrix or Coord
+    id::String # optional
+    index::int
+    coord::Coord # optional
+end
+
+struct TimeWindow
+    tw_start::Float64
+    tw_end::Float64
+end
+
+struct ResourcePrices
+    names::Vector{String} # fixed_cost_coef, travel_dist_coef,  travel_time_coef, service_time_coef,  waiting_time_coef
+    coef::Vector{Float64}
+end
+
+struct ResourceConsumptions
+    names::Vector{String} #  travel_dist_conso,  travel_time_conso, service_time_conso,  waiting_time_conso, vehicle_type_id
+    conso::Vector{Float64}
+end
+
+struct Depot <: AbstractNode
+    location::Location
+    time_windows::Vector{TimeWindow} # optional
+end
+
+struct Operation <: AbstractNode
+    location::Location
+    conso::ResourceConsumptions
+    time_windows::Vector{TimeWindow} # optional
+    req_index::Int
+    operation_type::String # Pickup, Delivery, Shipment, Cleaning, ...
+end
+
+struct Request
+    id::String
+    index::Int
+    operations::Vector{Operation} # Sequence of operations ; can be limited to one; can be a pair of Pickup and Delivery, or a triplets including a cleaning first, etc
+end
+
+struct VehicleBaseType
+    prices::ResourcePrices
+    capacity::ResourceConsumptions

```

Comment:
It is not intuitive to put capacity under the type ResourceConsumptions

Reply:
we can have this: `const ResourceValues = Dict{RESOURCE,Float64}` and use the name of the field of each structure to say whether it is a price, a capacity, an initial state or a consumption